sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -298,15 +315,25 @@ class GpuParquetWriter(
     dataSchema: StructType,
     compressionType: CompressionType,
     dateTimeRebaseException: Boolean,
+    int96RebaseException: Boolean,
     context: TaskAttemptContext)
   extends ColumnarOutputWriter(path, context, dataSchema, "Parquet") {
 
   val outputTimestampType = conf.get(SQLConf.PARQUET_OUTPUT_TIMESTAMP_TYPE.key)
 
   override def scanTableBeforeWrite(table: Table): Unit = {
-    if (dateTimeRebaseException) {
-      (0 until table.getNumberOfColumns).foreach { i =>
-        if (RebaseHelper.isDateTimeRebaseNeededWrite(table.getColumn(i))) {
+    (0 until table.getNumberOfColumns).foreach { i =>
+      val col = table.getColumn(i)
+      // if col is a day
+      if (dateTimeRebaseException && col.getType == DType.TIMESTAMP_DAYS) {
+        if (RebaseHelper.isDateRebaseNeeded(col)) {
+          throw DataSourceUtils.newRebaseExceptionInWrite("Parquet")
+        }
+      }
+      // if col is a time
+      else if (col.getType.hasTimeResolution && (int96RebaseException ||
+          outputTimestampType != ParquetOutputTimestampType.INT96 && dateTimeRebaseException)) {
+        if (RebaseHelper.isTimeRebaseNeeded(col)) {
```

Comment:
Related to the previous comments, the rebase helper method already checks the column types.  If we make the boolean track whether or not we need to check timestamp rebasing, this then would simplify to:
```scala
      else if (timestampRebaseException && RebaseHelper.isTimeRebaseNeeded(col)) {
```